pilot/cmd/pilot-agent/status/ready/probe_test.go

```diff
@@ -87,7 +86,7 @@ server.state: 0`,
 
 	for _, tt := range cases {
 		t.Run(tt.name, func(t *testing.T) {
-			server := createAndStartServer(tt.stats)
+			server := testserver.CreateAndStartServer(tt.stats, "127.0.0.1:1234")
```

Comment:
nit: I know we don't do it today, but is it possible to use `:0` so the port is dynamic?

Reply:
yes we could but the caller checks stats on this port so we will have to return that some how back.

Reply:
couldn't you do something like `server.Addr().Port()` or equivilent?

Reply:
Yes .We could. I will change it

Reply:
https://github.com/istio/istio/pull/31901